pkg/location/extensions/extensions.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"context"
+	"fmt"
+)
+
+type Reference interface {
+	fmt.Stringer
+	Type() string
+	Validate() error
+}
+
+type ReferenceLock interface {
+	Reference
+}
+
+type IdentifierGetter interface {
+	GetIdentifier() (string, bool)
+}
+
+type LockGetter interface {
+	GetLock() (string, bool)
```

Comment:
should this return `location.ReferenceLock` ?

Reply:
That's a good question, but I think the problem is that the name "GetLock" is very bad.

The value being returned here is the Digest string, or Commit string, or for a custom the appropriate string that uniquely identifies the immutable revision of remote content.

So at the moment to turn a Reference to a ReferenceLock - you call:
`refLock ReferenceLock, err error := mutate.Lock(ref Reference, lock string)`

And to get the lock string back out of a ReferenceLock - you call:
`lock string, err error := location.Lock(refLock)`

_**So**_ I think the `lock string` needs a better name for digest-or-commit value. If that was called a `Foo` then the above pair would change to:

`refLock ReferenceLock, err error := mutate.Lock(ref Reference, foo string)`
and
`foo string, err error := location.GetFoo(refLock)`

Reply:
There was a another thread which @droot also mentioned Address and ResolvedAddress as names. I think that was for some Ref RefLock properties, not the Ref/RefLock interface names themselves --- but if there is a better name now could be a good time to change